modules/search/search.go

```diff
@@ -0,0 +1,166 @@
+package search
+
+import (
+	"github.com/bwmarrin/discordgo"
+	"github.com/lordralex/absol/api"
+	"github.com/lordralex/absol/api/database"
+	"github.com/lordralex/absol/api/logger"
+	"github.com/spf13/viper"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Module struct {
+	api.Module
+}
+
+// Load absol commands API
+func (*Module) Load(ds *discordgo.Session) {
+	api.RegisterCommand("s", RunCommand)
+	api.RegisterCommand("search", RunCommand)
+
+	api.RegisterIntentNeed(discordgo.IntentsGuildMessages, discordgo.IntentsDirectMessages)
+}
+
+func RunCommand(ds *discordgo.Session, mc *discordgo.MessageCreate, _ string, args []string) {
+	if len(args) == 0 {
+		err := SendWithSelfDelete(ds, mc.ChannelID, "You must specify a search string!")
+		if err != nil {
+			return
+		}
+		return
+	} else if len(strings.Join(args, "")) < 3 {
+		err := SendWithSelfDelete(ds, mc.ChannelID, "Your search is too short!")
+		if err != nil {
+			return
+		}
+		return
+	}
+
+	max := viper.GetInt("factoids.max")
+	if max == 0 {
+		max = 5
+	}
+
+	db, err := database.Get()
+	if err != nil {
+		err = SendWithSelfDelete(ds, mc.ChannelID, "Failed to connect to database")
+		logger.Err().Printf("Failed to connect to database\n%s", err)
+		return
+	}
+
+	// gets the factoids table
+	var factoidsList []factoid
+	err = db.Find(&factoidsList).Error
+	if err != nil {
+		return
+	}
+
+	pageNumber := 0
+	pageNumber, err = strconv.Atoi(args[len(args)-1]) // if the last arg is a number use it as the page number
```

Comment:
Gorm can handle paging for you as well

Reply:
Is there a reason having Gorm do it would be better?

Reply:
Because gorm runs it on the database, where it's faster + optimized + isn't a memory hog, unlike this.